### ISSUE TITLE
Inconsistent use of angle units (degrees vs radians) in polar axes

### ISSUE THREAD

**Issue by dww100** [matplotlib/matplotlib#4986]
When making a polar plot the units used for input data are radians, however the axes produced for the plot are in degrees. This seems unnecessarily confusing. A short illustrative example is provided below:

```
import matplotlib.pyplot as plt
theta = [1.5, 3, 10, 90, 180]
r = range(1,6)
ax = plt.subplot(111,polar=True)
ax.scatter(theta,r)
plt.show()
```

![polar-rad-deg-issue](https://cloud.githubusercontent.com/assets/4116562/9463130/9597b33a-4b12-11e5-876d-250371a32de5.png)

This extends to the placing of axis labels etc, i.e. to move the r axis labels to the vertical grid line you use degrees:

```
ax.set_rgrids(r, angle = 90)
```

As mentioned to @jenshnielsen

**Comment by jenshnielsen** [comment]
Im not sure this is in scope for 2.0 but in the long run I think the issues in #4897 shows clearly that the unit system needs some work and this should include issues like this one IMHO

**Comment by timhoffm** [comment]
This is really a mess. Radians/degrees seem to be chosen randomly:

**Radians:**
- internally used coordinates and given data points
- positional arguments in `set_thetalim`
- `get/set_theta_offset`

**Degrees:**
- keyword arguments in `set_thetalim`
- `get/set_thetamin/max`
- parameter `offset` in `set_theta_zero_location`
- `set_thetagrids`

I'm not even sure which API I would like to have. Does anybody have thoughts on this? Maybe we can then come up with a reasonable migration strategy, e.g. with warnings depending on data with typical ranges ~ -2pi:pi vs -360:360.

**Comment by jklymak** [comment]
This is part of a "project", but I agree this API problem is pretty urgent!

I would *guess* most folks would prefer degrees, but?

**Comment by anntzer** [comment]
My personal preference would be for radians, because that matches what numpy uses basically everywhere (even in things such as np.unwrap, which are not bound by consistency with C functions (sin, cos, etc.)).  Also it would be quite hard to really insulate the internals which for now are, well, using radians.  OTOH I agree anything consistent, whether degrees everywhere or radians everywhere, would be nicer than the current mess.

**Comment by timhoffm** [comment]
I agree that changing everything to radians is technically simpler. The only concern I have here is that the axis is labelled in degrees, which would still be somewhat awkward.

**Comment by gewoondim1** [comment]
Would it be possible to "set" matplotlib to degree or radian mode?

**Comment by timhoffm** [comment]
> Would it be possible to "set" matplotlib to degree or radian mode?

AFAIR we've refrained from such approaches because of the added complexity. Also, it wouldn't make the interface more consistent unless you fully implement both versions, which is more work than even coercing to one choice of angles.

**Comment by dstansby** [comment]
I'd be pro going (or at least trying to go, and having as a policy) all radians.

**Comment by QuLogic** [comment]
Some of this inconsistency is due to inadvertent clashes during subsequent refactoring. Historically, the various forms were consistent in their own manner, though maybe not overall. It's true that the plotting routines use radians, but _not_ `PolarAxes` direct API.
* tick labels are in degrees
* `thetagrids` has since forever used degrees
* `theta_offset` is pseudo-internal so it's in radians (this is only my reading of it, I don't know the original intention)

When I added wedge limits, I followed the theta-is-degrees as far as `PolarAxes`'s public API was concerned (i.e., `offset` in `theta_zero_location`, `thetalim`). Because "x" was internal, it was _supposed_ to be in radians.

A later PR merged `thetalim` and `xlim` (similar for `r`) because they were redundant, but this forgot the fact that they were in different units (and I'm fairly certain I reviewed that, so admittedly, I forgot about it too.) This makes the differing units much more apparent since they appear in the same function call.

Since it's basically impossible to differentiate between radians and degrees, I'm not sure how to converge on one unit gracefully, other than adding new API (or restoring the old split).

**Comment by timhoffm** [comment]
> Since it's basically impossible to differentiate between radians and degrees, I'm not sure how to converge on one unit gracefully, other than adding new API (or restoring the old split).

I assume that radiant will typically be in the range -pi to 2pI. Degrees will typically be 0 or their absolute value will be >=10. Maybe this is enough for a heuristic generating A warning.

**Comment by QuLogic** [comment]
That might have worked with only theta limits, because limits of ~pi would produce an extremely thin axes. However, with r limits as well, you could make an axes using those ~pi limits without it being a mistake and without it being totally squashed. The heuristic would need to be a little more complicated.
